test/graphql/plug/endpoint_test.exs

```diff
@@ -46,10 +47,23 @@ defmodule GraphQL.Plug.EndpointTest do
     assert_query {:post, "/", query: "{greeting}"}, {200, success}
   end
 
+  test "root data can be set at request time" do
+    defmodule TestRootPlug do
+      use Plug.Builder
+      def root_eval(_conn), do: %{greeting: "Root"}
+      plug GraphQL.Plug.Endpoint, [schema: {TestSchema, :schema}, root: &TestRootPlug.root_eval/1 ]
```

Comment:
The callback needs to be in the same `{module, function}` format as the schema otherwise Phoenix code reloading messes with it.